astropy/io/misc/hdf5.py

```diff
@@ -266,11 +279,18 @@ def write_table_hdf5(table, output, path=None, compression=False,
     if serialize_meta:
         header_yaml = meta.get_yaml_from_table(table)
 
-        try:
-            dset.attrs[META_KEY] = [h.encode('utf8') for h in header_yaml]
-        except Exception as e:
-            warnings.warn("Attributes could not be written to the output HDF5 "
-                          "file: {0}".format(e))
+        header_encoded = [h.encode('utf-8') for h in header_yaml]
+        if compatibility_mode:
```

Comment:
Ah yes, the decorator only handles changing keywords, not the odd corner case of adding a keyword that is immediately deprecated...